add support for --hod-module + some refactoring + docs update

boegel opened the pull request:
@ehiggs: This enables me to submit to golett from the login nodes, using the system Python, as follows:

```
hod create --workdir=$VSC_SCRATCH/hod --dist=IPython-notebook-3.2.1 --hod-module=hanythingondemand/3.0.0dev-intel-2015a-Python-2.7.10
```

Seems to be working (I can connect to the IPython web UI for a cluster that was submitted like this).

boegel committed:
add support for --hod-module, remove hackery to automagically determine HOD module to load

boegel committed:
make netaddr and netifaces optional (only needed in an HOD job)

boegel committed:
move check whether --hod-module is defined into PbsHodJob constructor

boegel committed:
update docs for --hod-module + reorganize to avoid duplication

boegel committed:
more doc tweaks

boegel committed:
minor style fix

boegel committed:
fix options.options typo

ehiggs commented on hod/rmscheduler/hodjob.py:
```diff
@@ -118,28 +118,13 @@ class PbsHodJob(MympirunHod):
+            self.modules = [options.options.hod_module]
+        else:
+            sys.stderr.write("HOD module to load in job is not specified (--hod-module).\n")
+            sys.exit(1)
```
Can we not throw here?

boegel replied:
if it's OK to require `--hod-module` for `genconfig`, yes, this can move to `validate_pbs_option`

ehiggs commented on hod/node/node.py:
```diff
@@ -41,17 +39,49 @@
 
 from hod.commands.command import ULimit
 
+try:
```
This again!?

boegel replied:
if we want `hod create` and `hod batch` to work with only the system Python on our systems, we need this

unless we install `netifaces` and `netaddr` just to make the `import's work, which is stupid imho since `netifaces` and `netaddr` are only needed in the HOD 'backend', not for interactive use of `hod`

ehiggs replied:
Sure. Then let's lift all these decorator checks into a common module checker, if possible.

ehiggs commented on docs/Command_line_interface.rst:
```diff
@@ -117,22 +95,67 @@ Create a hanythingondemand cluster, with the specified label (optional) and clus
+**must be specified**
+
+Specify the ``hanythingondemand`` module that must be loaded in the job that is submitted for the HOD cluster;
+can also be specified via ``$HOD_CREATE_HOD_MODULE``.
```
I want to see an associated EasyBuild file that sets this so it's always in the environment when someone's using it.

boegel replied:
I'm planning exactly this: the module we'll put in place for interactive use on the login node should define `$HOD_CREATE_HOD_MODULE` and `$HOD_CREATE_WORKDIR` (+ the same for the `batch` subcommand)

this will make it a lot easier for users

Commit:
Add --hod-module to create and batch tests.

Commit:
Move check for `--hod-module` from hodjob to hod.option

Since that's where cmd line parting is done.

ehiggs commented:
https://github.com/boegel/hanythingondemand/pull/6 addresses my comments and fixes tests.

Commit:
Split out genconfig's validation into another function.

There is some copy paste here, but trying to remove the copy paste is difficult
since it's more than just 'make sure all these params are there' due to the xor
on hodconf and dist.

Commit:
Derp derp derp derp. I am an idiot. Thanks @boegel.

boegel committed:
Merge pull request #6 from ehiggs/hod_module

Fix hod_module tests and move cmd line verification to cmd line verification function.
